apps/public-docsite-v9/src/Concepts/Slots/Slots.stories.mdx

```diff
@@ -333,27 +358,52 @@ const useButtonStyles_unstable = (state: ButtonState)
 Hooks like `useButtonStyles` create classes and conditionally apply them based on the input state. Classes are typically created and applied with `makeStyles` and `mergeClasses`.
 For instance, if a component is disabled then disabled styles are added.
 
+### Declaring components slots in the state
+
+The `Button` component defines its slots `root` and `icon` in the state.
+
+```ts
+const useButton_unstable = (props: ButtonProps, ref: React.Ref<HTMLButtonElement | HTMLAnchorElement>): ButtonState => {
+  return {
+    root: slot.always({ ...props, ref }, { elementType: 'button' }),
+    icon: slot.optional(props.icon, { elementType: 'span' }),
+  };
+};
+```
+
+`slot.always` creates a slot that will always render, and as such the user may not provide `null` to opt-out of this slot (NonNullable slot). `slot.optional` creates a slot that can be opted out and is not rendered by default, it only renders if `props.icon` is different from `undefined` (Optional slot).
+
+Both `slot.always` and `slot.optional` methods will create a slot definition that can be consumed by the render method to properly render a slot, these methods ensure the local logic provided by the state hook will remain on the slots internals.
+
+### Rendering components with slots
+
 - `renderButton` takes in `ButtonState` and renders content.
 
 ```tsx
+/** @jsxRuntime classic */
+/** @jsx createElement */
```

Comment:
I personally don't want to expose old pragma, but as the example exists. Let's do it 😿

Reply:
I'll see to update the doc once again once we got the pragma removed with this https://github.com/microsoft/fluentui/pull/28810